docs/interfacing_with_surrogates.rst

```diff
@@ -0,0 +1,250 @@
+
+Consider a PyTorch neural network,
+
+.. code-block:: python
+
+    import torch
+
+    class PyTorchMLP(torch.nn.Module):
+      def __init__(self, hidden_dim: int, n_hidden: int, output_dim: int, input_dim: int):
+        super().__init__()
+        self.model = torch.nn.Sequential(
+          torch.nn.Linear(input_dim, hidden_dim),
+          torch.nn.ReLU(),
+          *[torch.nn.Sequential(
+            torch.nn.Linear(hidden_dim, hidden_dim),
+            torch.nn.ReLU()
+          ) for _ in range(n_hidden)],
+          torch.nn.Linear(hidden_dim, output_dim)
+        )
+
+      def forward(self, x):
+        return self.model(x)
+
+    torch_model = PyTorchMLP(hidden_dim, n_hidden, output_dim, input_dim)
+
+This model can be converted to a Flax model as follows:
+
+.. code-block:: python
+
+    from flax import linen
+
+    class FlaxMLP(linen.Module):
+      hidden_dim: int
+      n_hidden: int
+      output_dim: int
+      input_dim: int
+
+    @linen.compact
+    def __call__(self, x):
+      x = linen.Dense(self.hidden_dim)(x)
+      x = linen.relu(x)
+      for _ in range(self.n_hidden):
+        x = linen.Dense(self.hidden_dim)(x)
+        x = linen.relu(x)
+      x = linen.Dense(self.output_dim)(x)
+      return x
+
+    flax_model = FlaxMLP(hidden_dim, n_hidden, output_dim, input_dim)
+
+As this is only the model architecture, we need to load the trained weights separately.
+This can be a bit fiddly as you have to map from the parameter names in the weights checkpoint file to the parameter names in the Flax model.
+
+For loading weights from a PyTorch checkpoint, you might do something like:
+
+.. code-block:: python
+
+    import torch
+
+    state_dict = torch.load(PYTORCH_CHECKPOINT_PATH)
+
+    params = {}
+    for i in range(n_hidden_layers):
+      layer_dict = {
+        "kernel": jnp.array(
+          state_dict[f"model.{i*2}.weight"]
+        ).T,
+        "bias": jnp.array(
+          pytorch_state_dict[f"model.{j*2}.bias"]
+        ).T,
+      }
+      params[f"Dense_{i}"] = layer_dict
+
+    params = {'params': params}
+
+
+The model can then be called like any Flax model,
+
+.. code-block:: python
+
+    output_tensor = flax_model.apply(params, input_tensor)
+
+
+Option 2: converting a Pytorch model to a JAX model
+===================================================
+
+.. warning::
+    The `torch_xla2`_ package is still evolving, which means there may be unexpected breaking changes. Some of the methods described in this section may become deprecated with little warning.
+
+If your model is in PyTorch, you could also consider using the `torch_xla2`_ package to do the conversion to JAX automatically.
+
+.. code-block:: python
+
+    import torch
+    import torch_xla2 as tx
+
+    trained_model = torch.load(PYTORCH_MODEL_PATH, weights_only=False) # Use weights_only=False if you want to load the full model
+    params, jax_model_from_torch = tx.extract_jax(model)
```

Comment:
I think it needs be be jitted (build in good practice: otherwise many users might be hit by terrible performance)